### DISCUSSION TITLE
Add new method to an existing class

### DISCUSSION THREAD

**Discussion by maarsalien** [vfsfitvnm/frida-il2cpp-bridge#334] [General]
Hi, i am working on a project where i need an ``OnGUI`` method, i checked in all the class from all assembly, there are some but all the classes are not active class mean the class is not used by the game, so i wonder how can i push my custom OnGUI method to an existing class, is it possible ?

**Comment by vfsfitvnm** [comment]
Nope

**Reply by maarsalien** [reply]
Thanks

**Reply by vfsfitvnm** [reply]
More in general, IL2CPP deosn't support JIT compilation.

I think you could understand what `OnGui` does first (I don't know) and somehow manage to replicate that behavior without relying on runtime objects.

I have zero Unity knowledge, so unfortunately I can't help you further :/

**Reply by maarsalien** [reply]
From what i have learned so far, all it take is to have a method caled OnGUI, this event fire automatically from any active class during unity render time, and this is the only way to use Unity IMGUI component, they (UI Component) dont render outside of OnGUI
